packages/compass-data-modeling/src/store/diagram.ts

```diff
@@ -256,28 +405,28 @@ export function moveCollection(
 }
 
 export function applyEdit(
-  rawEdit: Omit<Edit, 'id' | 'timestamp'>
-): DataModelingThunkAction<void, ApplyEditAction | ApplyEditFailedAction> {
+  rawEdit: EditAction
+): DataModelingThunkAction<boolean, ApplyEditAction | ApplyEditFailedAction> {
   return (dispatch, getState, { dataModelStorage }) => {
     const edit = {
       ...rawEdit,
       id: new UUID().toString(),
       timestamp: new Date().toISOString(),
-      // TS has a problem recognizing the discriminated union
-    } as Edit;
+    };
     const { result: isValid, errors } = validateEdit(edit);
     if (!isValid) {
       dispatch({
         type: DiagramActionTypes.APPLY_EDIT_FAILED,
         errors,
       });
-      return;
+      return isValid;
```

Comment:
Returning the value from action so that other actions using it can adjust their behavior if needed, otherwise there's no indication that the action failed. There is currently no validation in the form and strictly speaking we are fully in control there so we can't actually fail this validation, but it might be useful in the future